src/data_processing/base_datamodule.py

```diff
@@ -0,0 +1,21 @@
+from abc import ABC, abstractmethod
+from typing import Tuple, List
+import numpy as np
+
+class BaseDataModule(ABC):
+    """Abstract interface that supplies multiple datasets as (X_list, y_list)."""
+
+    def __init__(self, random_seed: int = 42):
```

Comment:
It feels off to have a random seed here.  
There are for sure data modules without randomness.  

I recommend to delete this and let the user decide the constructor signature.